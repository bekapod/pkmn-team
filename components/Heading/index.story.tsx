import { storiesOf } from "@storybook/react";
import React from "react";
import Heading from ".";

storiesOf("Heading", module).add(
  "default",
  (): JSX.Element => <Heading>Cupcake ipsum dolor sit amet</Heading>
);
