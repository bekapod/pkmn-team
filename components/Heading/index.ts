import styled from "styled-components/macro";
import * as variables from "../../helpers/variables";

const Heading = styled.h1`
  margin-top: 0;
  padding: ${variables.spacing.md}px ${variables.gutters.page}px;
  color: ${variables.colors.white};
  background-color: ${variables.colors.secondary};
`;

export default Heading;
