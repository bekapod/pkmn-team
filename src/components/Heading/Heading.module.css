.base {
  margin-top: 0;
  margin-bottom: var(--spacing-4);
  padding: var(--spacing-4) var(--spacing-4);
  color: var(--color-white);
  background-color: var(--color-sherpa-blue-800);

  @media (--medium-viewport) {
    padding: var(--spacing-4) var(--spacing-6);
  }
}
