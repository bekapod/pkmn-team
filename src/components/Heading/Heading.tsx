import styled from 'styled-components/macro';
import { media } from '~/lib/media';

export const Heading = styled.h1`
  margin-top: 0;
  margin-bottom: var(--spacing-lg);
  padding: var(--spacing-md) var(--gutter-page);
  color: var(--color-white);
  background-color: var(--color-secondary);

  ${media.medium`
    padding: var(--spacing-md) var(--gutter-page-md);
  `}
`;
