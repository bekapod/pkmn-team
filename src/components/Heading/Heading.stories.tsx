import { Meta } from '@storybook/react/types-6-0';
import { Heading } from './Heading';

export default {
  title: 'Components/Heading',
  component: Heading
} as Meta;

export const Standard = (): JSX.Element => (
  <Heading>Cupcake ipsum dolor sit amet</Heading>
);
