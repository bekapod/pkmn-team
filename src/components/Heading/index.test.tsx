import React from "react";
// tslint:disable-next-line:no-implicit-dependencies
import { render } from "react-testing-library";
import Heading from ".";

describe("<Heading />", () => {
  it("renders an h1 with children", () => {
    const { getByText } = render(<Heading>This is a heading</Heading>);

    expect(getByText(/This is a heading/).tagName).toBe("H1");
  });
});
