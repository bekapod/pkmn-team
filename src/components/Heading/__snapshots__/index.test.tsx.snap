// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> renders without crashing 1`] = `
<h1
  className="Heading-b3adig-0 jwQDLL"
>
  This is a heading
</h1>
`;
